base_images/debian_base-setup.sh

```diff
@@ -150,6 +150,11 @@ $SUDO chmod 755 /usr/bin/version_find_latest
 timebomb 20240330 "prevent us from getting broken tar-1.35+dfsg-3"
 ( set -x; $SUDO apt-mark hold tar; )
 
+# 2024-03-27 grub 2.12-1+b1 causes boot failures, "bli.mod not found". Block it.
```

Comment:
(Non-blocking) Did you happen to find anything online about this, that might be good to include in the comment?

I'm considering a frustrated "not Ed" maintainer coming along with a PR and not really knowing what to do with this timebomb (or any of them, really).  Would you consider (now or later) including comments on how to verify these?  I'm fine if that's simply: "Ask Ed" or a more complete/complex action-item.  Just something so people aren't left "hanging", wondering what to do.

Reply:
I found nothing, and didn't even bother looking. We already know for a fact that debian and google do not play well together (see the other grub timebomb); it doesn't seem useful to figure out why.

What _does_ bother me a lot is that there's no way (that I know of) to see if there's an update to the buggy version. Like, the `tar` block, maybe debian has fixed the bug and pushed a new version, but there's not really any way to look at the logs and see "oh hey there's a version beyond the timebomed one, let's see if that fixes things". I've just added a TODO item for myself to see if there's a debian way to block one specific version, instead of blocking all upgrades.

Anyhow, [IT WORKED!](https://cirrus-ci.com/task/5003396384555008). Now to see what else fails. If it passes podman and buildah CI, I'll address your other (very welcome, thank you) feedback.

Reply:
Yay :tada: 

Renovate would be the tool for watching versions, it can accept a regex search pattern to locate the necessary "in use" version details.  The challenge is on the back-end, getting the "latest" versions of things that are available.  There's [a versioning module called repology](https://docs.renovatebot.com/modules/datasource/repology/), and it may work well for Debian. But the last time I tried, it fell flat on everything Fedora related :disappointed:  Anyway, there may be opportunities in this space, given the time/effort required to configure it.  Bother me later if you think this is really important and should be pursued further.

Reply:
Nah, already solved it, with a `/etc/apt` thing that failed because I forgot to SUDO and I'm not going to bother re-pushing today anyway.

Reply:
I always forget the `$SUDO` also :frowning_face: